### PR TITLE
Calculate 5% cashback amount

### PR DESCRIPTION
**Pull request by earlyshen**
gcmall

### CODE REVIEW

File: balchecker.py [hunk 1]
```diff
@@ -20,9 +20,11 @@ def getBalance(cardInfo):
         lastFour = match[0][1]
         availableBalance = match[1][1]
         initialBalance = match[2][1]
-        return {'lastFour': lastFour, 'availableBalance': availableBalance, 'initialBalance': initialBalance}
+        cashbackMatch = re.findall(r"INTELISPEND - EGIFT.+\"textRightAlign\">\$([\d.]+)", response.text)
```
Comment (chinghanyu) [inline balchecker.py]: I have four cards and each of them was given cash back twice and this script catches only the first one.

Comment (earlyshen) [inline balchecker.py]: Interesting could you attach the HTML response? Just the table part

**Comment by earlyshen** [comment]
@chinghanyu Just updated the PR. I don't have the card with CSR balance override so please check whether this works or not. Thanks :)

File: balchecker.py [hunk 2]
```diff
@@ -22,7 +22,9 @@ def getBalance(cardInfo):
         initialBalance = match[2][1]
         cashbackMatch = re.findall(r"INTELISPEND - EGIFT.+\"textRightAlign\">\$([\d.]+)", response.text)
         cashback = reduce(lambda x, y: x+float(y), cashbackMatch, 0.0) if len(cashbackMatch) > 0 else 0.0
-        return {'lastFour': lastFour, 'availableBalance': availableBalance, 'initialBalance': initialBalance, 'cashback': "${0:.2f}".format(cashback)}
+        csrOverrideMatch = re.findall(r"<td>\s*<\/td><td Class=\"textRightAlign\">\$([\d.]+)", response.text)
```
Comment (chinghanyu) [inline balchecker.py]: Well. This piece of code is not working with my case. I will check it later.

Comment (earlyshen) [inline balchecker.py]: Found the issue. It is not white space, its `&nbsp;`

**Comment by earlyshen** [comment]
gcmall
@chinghanyu Check this out :)

File: balchecker.py [hunk 3]
```diff
@@ -70,8 +74,8 @@ def validateCard(row):
 
         balance = getBalance(cardInfo)
         if balance.get('lastFour') == '-1':
-            print '{:>13} {:>19}'.format(cardInfo.get('CardNumber')[-4:], 'Card not found')
+            print '{:>6} {:>29}'.format(cardInfo.get('CardNumber')[-4:], 'Card not found')
```
Comment (chinghanyu) [inline balchecker.py]: Might need to adjust the position of `Card not found` message.

File: balchecker.py (hunk 1 above)
Comment (earlyshen) [inline balchecker.py]: The `reduce` part in next line is already take care of multiple transactions. 
I agree that we need to refactor this when we have the `Transaction` class

**Pull request merged**